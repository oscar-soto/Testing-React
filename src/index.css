html, body {
  padding: 70px;
  color: #FFF;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  background-color: #21232A;
}

button {
  padding: 10px 20px;
  border-radius: .25rem;
  border: none;
  font-size: 1.25rem;
  transition: all ease-in-out 0.3s;
  cursor: pointer;
  margin-right: 10px;
}
button:hover {
  opacity: 0.8;
}