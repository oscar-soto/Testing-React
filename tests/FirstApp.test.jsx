import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Prueba en <FirsApp />', () => {
  test('Debe hacer match con el snapshot', () => {
    const title = 'Hola, soy oscar';
    const { container } = render(<FirstApp title={title} />);

    // console.log(container)

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el titulo en un h1', () => {
    const title = 'Hola, soy oscar';
    const { container, getByText } = render(<FirstApp title={title} />);

    expect( getByText(title) ).toBeTruthy();

    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toContain( title )
  });
});
