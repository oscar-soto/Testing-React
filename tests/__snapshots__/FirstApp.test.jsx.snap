// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirsApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    Hola, soy oscar
     
  </h1>
  <p>
    123
  </p>
  <p>
    Oscar Espinoza
  </p>
</div>
`;
